bee-network/bee-autopeering/src/config.rs

```diff
@@ -0,0 +1,329 @@
+        }
+    }
+
+    /// Turns the [`AutopeeringConfig`] into its TOML representation.
+    pub fn into_toml_config(self) -> AutopeeringConfigTomlBuilder {
+        AutopeeringConfigTomlBuilder {
+            enabled: self.enabled,
+            bind_addr: self.bind_addr,
+            entry_nodes: self.entry_nodes,
+            entry_nodes_prefer_ipv6: Some(self.entry_nodes_prefer_ipv6),
+            run_as_entry_node: Some(self.run_as_entry_node),
+            drop_neighbors_on_salt_update: Some(self.drop_neighbors_on_salt_update),
+        }
+    }
+}
+
+// Note: In case someone wonders why we use `Option<bool>`: Although serde actually provides a way to allow for the
+// default of a boolean parameter to be `true` - so that missing config parameters could be created on the fly - it felt
+// too awkward and also a bit too cumbersome to me: serde(default = "default_providing_function_name").
+
+/// The autopeering config JSON builder.
+///
+/// Note: Fields will be camel-case formatted.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+#[cfg_attr(test, derive(Eq, PartialEq))]
+#[serde(rename = "autopeering")]
+pub struct AutopeeringConfigJsonBuilder {
+    /// Wether autopeering should be enabled.
+    pub enabled: bool,
+    /// The bind address for the server.
+    #[serde(rename = "bindAddress")]
+    pub bind_addr: SocketAddr,
+    /// The entry nodes for bootstrapping.
+    #[serde(rename = "entryNodes")]
+    pub entry_nodes: Vec<AutopeeringMultiaddr>,
+    /// Whether `Ipv4` or `Ipv6` should be preferred in case a hostname supports both.
+    #[serde(rename = "entryNodesPreferIPv6")]
+    pub entry_nodes_prefer_ipv6: Option<bool>,
+    /// Whether the node should run as an entry node.
+    #[serde(rename = "runAsEntryNode")]
+    pub run_as_entry_node: Option<bool>,
+    /// Whether all neighbors should be disconnected from when the salts are updated.
+    #[serde(rename = "dropNeighborsOnSaltUpdate")]
+    pub drop_neighbors_on_salt_update: Option<bool>,
+}
+
+impl AutopeeringConfigJsonBuilder {
+    /// Builds the actual `AutopeeringConfig`.
+    pub fn finish(self) -> AutopeeringConfig {
+        AutopeeringConfig {
+            enabled: self.enabled,
+            bind_addr: self.bind_addr,
+            entry_nodes: self.entry_nodes,
+            entry_nodes_prefer_ipv6: self.entry_nodes_prefer_ipv6.unwrap_or(ENTRYNODES_PREFER_IPV6_DEFAULT),
+            run_as_entry_node: self.run_as_entry_node.unwrap_or(RUN_AS_ENTRYNODE_DEFAULT),
+            drop_neighbors_on_salt_update: self
+                .drop_neighbors_on_salt_update
+                .unwrap_or(DROP_NEIGHBORS_ON_SALT_UPDATE_DEFAULT),
+        }
+    }
+}
+
+impl Default for AutopeeringConfigJsonBuilder {
+    fn default() -> Self {
+        Self {
+            enabled: false,
+            bind_addr: SocketAddr::new(AUTOPEERING_BIND_ADDR_DEFAULT, AUTOPEERING_BIND_PORT_DEFAULT),
+            entry_nodes: Vec::default(),
+            entry_nodes_prefer_ipv6: Some(false),
```

Comment:
For this and following, use the constants ?

Reply:
oh yea. Of course :blush: